### PR TITLE
Add mask for entangling gates

### LINKED ISSUE #31: Support dropping of entangling gates
So far we only support dropping parameterised gates. But especially with a focus on lowering expressivity of a circuit it might make sense to also enable dropping entangling gates. This could be realised by adding another mask to our `MaskedCircuit` class.

### PR DESCRIPTION
**Pull request by eileen-kuehn**
This PR adds functionality to support removal of entangling gates from a given circuit via `Mask`s. Idea to support this feature is to explicitly support lowering the expressivity of a quantum circuit and therefore maybe raising the trainability.

Closes #31.

### CODE REVIEW

**Comment by eileen-kuehn** [comment]
@maxfischer2781, when reviewing, please have a specific look on how the `entangling_mask` is currently handled. I opted for making it optional what requires to test if it is set in case actions are performed. This will probably fail for the `execute` function if the mask is not properly initialised.
Any ideas on this? Further, I am a bit afraid, that this mask might even have not a so well-defined shape... However, I would like to give it a try already. What do you think?

File: maskit/masks.py [hunk 1]
```diff
@@ -293,6 +317,8 @@ def clear(self):
         self._layer_mask.clear()
         self._wire_mask.clear()
         self._parameter_mask.clear()
+        if self.entangling_mask:
```
Comment (maxfischer2781) [inline maskit/masks.py:320]:
```suggestion
        if self.entangling_mask is not None:
```

File: maskit/masks.py [hunk 2]
```diff
@@ -332,6 +358,10 @@ def copy(self: Self) -> Self:
         clone._wire_mask = self._wire_mask.copy(clone)
         clone.parameters = self.parameters.copy()
         clone.default_value = self.default_value
+        if self._entangling_mask:
```
Comment (maxfischer2781) [inline maskit/masks.py:361]:
```suggestion
        if self._entangling_mask is not None:
```

File: maskit/circuits.py [hunk 3]
```diff
@@ -24,8 +24,18 @@ def basic_variational_circuit(params, rotations, masked_circuit: MaskedCircuit):
             rotation(full_parameters[layer][wire], wires=wire)
 
         for wire in range(0, wires - 1, 2):
+            if (
+                masked_circuit.entangling_mask
```
Comment (maxfischer2781) [inline maskit/circuits.py:28]:
```suggestion
                masked_circuit.entangling_mask is not None
```

File: maskit/circuits.py [hunk 4]
```diff
@@ -24,8 +24,18 @@ def basic_variational_circuit(params, rotations, masked_circuit: MaskedCircuit):
...
+                and masked_circuit.entangling_mask[layer, wire]
+            ):
+                continue
             qml.CZ(wires=[wire, wire + 1])
         for wire in range(1, wires - 1, 2):
+            if (
+                masked_circuit.entangling_mask
```
Comment (maxfischer2781) [inline maskit/circuits.py:35]:
```suggestion
                masked_circuit.entangling_mask is not None
```

**Review by maxfischer2781 — CHANGES_REQUESTED** [review]
I've only sprinkled ``is not None``s on the cases that were relying on implicit falseyness. That's somewhat faster and should be more robust if you ever refactor.

Do you think you can hit the ``continue``s with unittests? I can't really comment on whether they work as expected.

**Comment by maxfischer2781** [comment]
Sorry, I missed your previous comment. TBH I'm not totally sure how exactly *this* change works, so it's hard for me to say how it might affect other things.

Perhaps we should discuss at a literal drawing board. :P

**Comment by eileen-kuehn** [comment]
> Do you think you can hit the `continue`s with unittests? I can't really comment on whether they work as expected.

I added another unit test that should exactly test for that :) Now you will have to review again!

**Review by maxfischer2781 — APPROVED** [review]
Pretty sneaky just raising the coverage while I was looking... 😜

**Pull request merged**